integration_tests/test_solo_machine.py

```diff
@@ -0,0 +1,182 @@
+        file.extractall(self.bin_dir)
+        file.close()
+        # check file exists
+        if not os.path.exists(self.bin_file):
+            raise Exception(f"solomachine binary file {self.bin_file} is not exists")
+        if not os.path.exists(self.sign_file):
+            raise Exception(
+                f"solomachine signature file {self.sign_file} is not exists"
+            )
+
+    def prepare(self):
+        self.download_bin_file()
+        self.get_chain_info()
+        self.set_env()
+
+    def run_sub_cmd(self, sub_cmd, json_output=True):
+        if json_output:
+            cmd = f"{self.bin_file} --output json {sub_cmd}"
+        else:
+            cmd = f"{self.bin_file} {sub_cmd}"
+        output = interact(cmd)
+        data = parse_output(output, json_output)
+        return data
+
+    def init(self):
+        sub_cmd = "init"
+        return self.run_sub_cmd(sub_cmd)
+
+    def add_chain(self):
+        sub_cmd = "chain add"
+        return self.run_sub_cmd(sub_cmd)
+
+    def connect_chain(self):
+        sub_cmd = f"ibc connect {self.chain_id}"
+        result = self.run_sub_cmd(sub_cmd, False)
+        if "error" in result:
+            raise Exception(result)
+
+    def get_balance(self):
+        sub_cmd = f"chain balance {self.chain_id} solotoken"
+        return self.run_sub_cmd(sub_cmd)["data"]["balance"]
+
+    # fixme after this PR merged, currently only support hex string amount
+    # https://github.com/crypto-com/ibc-solo-machine/pull/82
+    def mint(self, amount="0x20"):
+        sub_cmd = f"ibc mint {self.chain_id} {amount} solotoken"
+        return self.run_sub_cmd(sub_cmd)
+
+    # fixme fix after this PR merged, currently only support hex string amount
+    # https://github.com/crypto-com/ibc-solo-machine/pull/82
+    def burn(self, amount="0x10"):
+        sub_cmd = f"ibc burn {self.chain_id} {amount} solotoken"
+        return self.run_sub_cmd(sub_cmd)
+
+
+def test_solo_machine(cluster):
+    """
+    check solo machine
+    """
+    cli = cluster
+    base_port = cli.base_port(0)
+    wait_for_port(ports.grpc_port(base_port))
+    solo_machine = SoloMachine(base_port=base_port)
+    solo_machine.prepare()
+    data = solo_machine.init()
+    assert data["result"] == "success"
+    data = solo_machine.add_chain()
+    assert data["result"] == "success"
+    time.sleep(3)
+    solo_machine.connect_chain()
+    time.sleep(3)
+    balance_0 = int(solo_machine.get_balance())
+    assert balance_0 == 0
+    data = solo_machine.mint("0x20")
+    assert data["result"] == "success"
+    assert data["data"]["amount"] == "0x20"
+    balance_1 = int(solo_machine.get_balance())
+    assert balance_1 == 0x20
+    time.sleep(3)
+    data = solo_machine.burn("0x10")
+    assert data["result"] == "success"
+    assert data["data"]["amount"] == "0x10"
+    time.sleep(3)
+    balance_2 = int(solo_machine.get_balance())
+    assert balance_2 == balance_1 - 0x10
+    assert balance_1 > balance_2
```

Comment:
some of these assertions should be against the node directly, i.e. to check that the chain/cluster node indeed received it (if there's a bug in the solo machine CLI or if it connects to a wrong node)